content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+    };
+```
+### Example 7: Based on Example 2, with '--unique-idx'. Error Case.
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t1 register(Because The ConstantBuffer reserved for SRG1 will be bound to b0 register), and     will take over ALL1+. 
+        Texture2D<float4>        m_texSRV1;      // ERROR. There's no tN available to bind this resource to.
+        Texture2D<float4>        m_texSRV2;      // ERROR. There's no tN available to bind this resource to.
+        RWTexture2D<float4>      m_texUAV1;      // ERROR. There's no uN available to bind this resource to.
+    	RWTexture2D<float4>      m_texUAV2;      // ERROR. There's no uN available to bind this resource to.
+        Sampler                  m_sampler1;     // ERROR. There's no sN available to bind this resource to.
+        Sampler                  m_sampler2;     // ERROR. There's no sN available to bind this resource to.
+        ConstantBuffer<MyStruct> m_cb1; // ERROR. There's no bN available to bind this resource to.
+        ConstantBuffer<MyStruct> m_cb2; // ERROR. There's no bN available to bind this resource to.
+    };
+```
+### Example 8: Fix for Example 7.
+The solution to Example 7, is to move the declaration of `m_texSRV_unbounded` to the end of `SRG1`, basically after `m_cb2`.  
+Alternatively, you could create a second `SRG2`, and move the declaration of `m_texSRV_unbounded` to `SRG2` (As the last resource in `SRG2`, in case there are more resources in `SRG2`.)  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t2 register.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u3 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u4 register.
+        Sampler                  m_sampler1;     // Will be bound to s5 register.
+        Sampler                  m_sampler2;     // Will be bound to s6 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b7 register.
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b8 register.
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t9 register, and will take over ALL10+.
+    };
```

Comment:
```suggestion
ShaderResourceGroupSemantic slot1
{
    FrequencyId = 1;
};

struct MyStruct
{
    float3 m_a;
    float3 m_b;
};
    
// For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
// The ConstantBuffer reserved for SRG1 binds to b0 register.
ShaderResourceGroup SRG1 : slot1
{
    Texture2D<float4>        m_texSRV1;             // Binds to t1 register.
    Texture2D<float4>        m_texSRV2;             // Binds to t2 register.
    RWTexture2D<float4>      m_texUAV1;             // Binds to u3 register.
    RWTexture2D<float4>      m_texUAV2;             // Binds to u4 register.
    Sampler                  m_sampler1;            // Binds to s5 register.
    Sampler                  m_sampler2;            // Binds to s6 register.
    ConstantBuffer<MyStruct> m_cb1;                 // Binds to b7 register.
    ConstantBuffer<MyStruct> m_cb2;                 // Binds to b8 register.
    Texture2D<float4>        m_texSRV_unbounded[];  // Binds to t9 register, and takes over all 10+ registers.
};
```
* Moved comment: "(Because the ConstantBuffer reserved for….)" to simplify the comments for each resource declaration. 
* Remove indent. It's not necessary when defining within a code snippet (```cpp). This also allows longer comments to fit on the page.
* Simplified "will be bound to" to "binds to".
* Simplified "will take over" to "takes over"

Reply:
Fixed.